Userland/Libraries/LibJS/Runtime/Object.h

```diff
@@ -221,6 +221,8 @@ class Object : public Cell {
     static FlatPtr may_interfere_with_indexed_property_access_offset() { return OFFSET_OF(Object, m_may_interfere_with_indexed_property_access); }
     static FlatPtr indexed_properties_offset() { return OFFSET_OF(Object, m_indexed_properties); }
 
+    static FlatPtr has_magical_length_property_offset() { return OFFSET_OF(Object, m_has_magical_length_property); }
```

Comment:
This might belong as a youtube comment, but I was thinking that since you're adding [a magic property](https://github.com/SerenityOS/serenity/pull/21910/files#diff-13981cfc1a6f73100eb0ca226485436dbc4e0d4696f06de6cc25a3925d69278cR224) to `Array`, why not call it `isArray`, instead of `has_magical_length_property_offset`? Then you also have a clear fast path for `Array.isArray()` for ES5.

Reply:
That doesn't seem safe if the value is a `Proxy` object :thinking:

Reply:
ahh yes. I wrote my comment wrong. I actually meant the `m_has_magical_length_property = true;` property in the `Array` constructor. Sorry for the confusion.